topics/icrc_X_multiple_canister_call.md

```diff
@@ -0,0 +1,237 @@
+
+Response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "responses": [
+      {
+        "id": 1,
+        "result": {
+          "Ok": "16509194"
+        }
+      },
+      {
+        "id": 2,
+        "result": {
+          "Ok": "16509195"
+        }
+      }
+    ]
+  }
+}
+```
+
+#### Batch Call with Error
+
+Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrcX_batch_call_canisters",
+  "params": {
+    "mode": "",
+    "sender": "b7gqo-ulk5n-2kpo7-oalt7-p2kyl-o4j5l-kiuwo-eeybr-dab4l-ur6up-pqe",
+    "isAnonymous": false,
+    "requests": [
+      {
+        "id": 1,
+        "params": {
+          "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+          "method": "transfer",
+          "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+        }
+      },
+      {
+        "id": 2,
+        "params": {
+          "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+          "method": "transfer",
+          "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+        }
+      },
+      ,
+      {
+        "id": 3,
+        "params": {
+          "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+          "method": "transfer",
+          "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+        }
+      }
+    ]
+  }
+}
+```
+
+Response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "responses": [
+      {
+        "id": 1,
+        "result": {
+          "version": "1",
+          "Ok": "16509194"
+        }
+      },
+      {
+        "id": 2,
+        "result": {
+          "Err": {
+            "InsufficientFunds": {
+              "balance": {
+                "e8s": "0"
+              }
+            }
+          }
+        }
+      },
+      {
+        "id": 3,
+        "error": {
+          "code": 1000,
+          "message": "Network error: Unable to establish a connection to the server"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Errors
+
+In addition to the errors defined in [ICRC-25](./icrc_25_signer_interaction_standard.md#errors-3) this standard defines the following errors:
+
+
+| Code  | Message                                    | Meaning                                                                                          | Data                                                                                |
+|-------|--------------------------------------------|--------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------|
+| 10101 | Not processed due to batch request failure | The message was not processed as one of the preceding request in the batch resulted in an error. | (optional) Error details: <ul> <li>`message` (`text`, optional): message</li> </ul> |
```

Comment:
Error codes have been [updated in ICRC-49](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_25_signer_interaction_standard.md#errors-3) a while back, I would recommend to use error code 1001 since it doesn't fit within 2xxx, 3xxx or 4xxx categories.